agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -109,10 +109,9 @@ def on_container_destroy(event)
     # @param [String] container_id
     # @param [String] overlay_cidr
     def attach_overlay(container)
-      if container.overlay_suffix == OVERLAY_SUFFIX
-        network_adapter.attach_container(container.id, container.overlay_cidr)
-      else
-        # override label for existing containers that may need to be migrated
+      if container.overlay_network.nil?
+        # overlay network migration for 0.16 compat
+        # override overlay network /12 -> /16 suffix for existing containers that may need to be migrated
```

Comment:
Fixed. I thought I rgrep'd for all of these; I confused the grid supernet with the old subnet :)